Allow configuration of custom command creator

tgardner opened the pull request:
Allow specifying an optional custom command creator when using the IHost & IHostBuilder extension methods.

jeremydmiller commented:
@tgardner Shamefully, I just saw this. You would need to make these changes as overloads because this is a breaking change otherwise. And I'd want to see at least some kind of smoke test

tgardner commented:
@jeremydmiller - Apologies, I'm equally as guilty. I've updated as overloads and added a basic smoke test. 
Happy to revise if you've got anymore feedback.

tgardner commented:
Fixed by #93
